daemon/MenuDaemon.vala

```diff
@@ -258,5 +262,78 @@ namespace Gala {
                 push_in = true;
             }, 3, Gdk.CURRENT_TIME);
         }
+
+        public void show_desktop_menu (int x, int y) throws DBusError, IOError {
+            if (desktop_menu == null) {
+                var change_wallpaper = new Gtk.MenuItem.with_label (_("Change Wallpaper…"));
+                change_wallpaper.activate.connect (() => {
+                    try {
+                        AppInfo.launch_default_for_uri ("settings://desktop/appearance/wallpaper", null);
+                    } catch (Error e) {
+                        warning ("Failed to open wallpaper settings: %s", e.message);
+                        var message_dialog  = new Granite.MessageDialog.with_image_from_icon_name (
+                            "Failed to Open Wallpaper Settings",
+                            "The error was: Could not launch Switchboard.",
+                            "dialog-warning",
+                            Gtk.ButtonsType.CLOSE
+                        );
```

Comment:
Since we get some error messaging from GLib if this fails, this would be a good time to include that:
```suggestion
                        );
                        message_dialog.show_error_details (e.message);
```